shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -265,40 +272,48 @@ public void run() {
-                sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
-                sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
-                sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
-                sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
-            }
+            return applicationContext
+                .getPackageManager()
+                .getApplicationInfo(applicationContext.getPackageName(), PackageManager.GET_META_DATA);
         } catch (PackageManager.NameNotFoundException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private static void initResources(Context applicationContext) {
-        Context context = applicationContext;
-        new ResourceCleaner(context).start();
+    /**
+     * Initialize our Flutter config values by obtaining them from the
+     * manifest XML file, falling back to default values.
+     */
+    private static void initConfig(@NonNull Context applicationContext) {
+        Bundle metadata = getApplicationInfo(applicationContext).metaData;
 
-        Bundle metaData = null;
-        try {
-            metaData = context.getPackageManager().getApplicationInfo(
-                    context.getPackageName(), PackageManager.GET_META_DATA).metaData;
+        if (metadata == null) {
+            return;
+        }
 
-        } catch (PackageManager.NameNotFoundException e) {
-            Log.e(TAG, "Unable to read application info", e);
+        sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
+        sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
+
+        sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
+        sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
+        sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
+        sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
+    }
+
+    private static void initResources(@NonNull Context applicationContext) {
+        // When the AOT blobs are contained in the native library directory, 
+        // we don't need to extract them manually because they are 
+        // extracted by the Android Package Manager automatically.
+        if (!sSnapshotPath.equals(PathUtils.getDataDirectory(applicationContext))) {
+            return;
         }
 
+        Context context = applicationContext;
```

Comment:
Why are we creating a new context variable here?

Reply:
This how the code is: https://github.com/flutter/engine/blob/0b69c54cf5254f2131f5d6511816042d4b05233b/shell/platform/android/io/flutter/view/FlutterMain.java#L314-L315

But I agree, I just removed it.